src/CMakeLists.txt

```diff
@@ -344,30 +344,33 @@ SET_TARGET_PROPERTIES(${CMAKE_PROJECT_NAME} PROPERTIES
 
 
 # Create a symbolic link from ${orig} to ${link}
+# If the orig and link point to the same thing, does nothing (-> in-source build)
 # If ${link} already exists, does nothing.
 function(make_symlink orig link)
 	# Get OS dependent path to use in `execute_process`
-	message("Creating symlink, orig = ${orig}; link = ${link}")
 	file(TO_NATIVE_PATH "${orig}" orig)
 	file(TO_NATIVE_PATH "${link}" link)
 
-	if (NOT EXISTS ${link})
-		if (CMAKE_HOST_UNIX)
-			set(command ln -s ${orig} ${link})
-		else()
-			if (IS_DIRECTORY ${orig})
-				set(command cmd.exe /c mklink /j ${link} ${orig})
+	if (NOT "${orig}" STREQUAL "${link}")
```

Comment:
This makes the nesting quite deep. Will an early return work instead?
```cmake
if (EXISTS "${link}")
	return()
endif()
```

Reply:
That's better, I couldn't find if cmake had returns. I'll rework this.